core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -170,7 +170,13 @@ public void onNext(HttpObject o) {
                 }
 
                 final HttpHeaders additionalHeaders = reqCtx.additionalResponseHeaders();
+                final HttpHeaders additionalTrailers = reqCtx.additionalResponseTrailers();
+
                 final ResponseHeadersBuilder newHeaders = fillAdditionalHeaders(headers, additionalHeaders);
+                if (endOfStream && !additionalTrailers.isEmpty()) {
+                    newHeaders.setIfAbsent(additionalTrailers);
+                }
+
                 if (newHeaders.contains(HttpHeaderNames.CONTENT_LENGTH) &&
```

Comment:
I'm a bit unsure what this logic is doing - does anyone know why does the presence of additional trailers affect the content length header?

Reply:
Yeah the logic was added in https://github.com/line/armeria/pull/1566
It's because curl does not get HTTP trailers when Content-length exists. (i.e Should send the response with chunked transfer encoding.)

Reply:
Ah - it's because trailers are only defined for chunked encoding, and armeria only applies chunked encoding if content-length isn't set. Added a comment to make that clearer.

Reply:
Yeah, thanks for the comment! 😄